knowledge/components/ShareButtons.tsx

```diff
@@ -0,0 +1,86 @@
+// import { useRouter } from "next/router";
+import CloseIcon from "@mui/icons-material/Close";
+import FacebookRoundedIcon from "@mui/icons-material/FacebookRounded";
+import LinkIcon from "@mui/icons-material/Link";
+import LinkedInIcon from "@mui/icons-material/LinkedIn";
+import TwitterIcon from "@mui/icons-material/Twitter";
+import { Box, IconButton, Snackbar, Tooltip } from "@mui/material";
+import { useRouter } from "next/router";
+import React, { useEffect, useState } from "react";
+
+export const ShareButtons = () => {
+  const router = useRouter();
+  const [url, setUrl] = useState("");
+  const [open, setOpen] = useState(false);
+
+  useEffect(() => {
```

Comment:
This `useEffect` is unnecessary
Just use
` const url = typeof window !== "undefined" ? window.location.href : "";`

Where you need it